drop-transfer/src/service.rs

```diff
@@ -117,6 +117,33 @@ impl Service {
         res
     }
 
+    pub async fn purge_transfers(&mut self, transfer_ids: Vec<String>) -> Result<(), Error> {
+        self.state
+            .storage
+            .purge_transfers(transfer_ids)
+            .await
+            .map_err(|_| Error::StorageError)
+    }
+
+    pub async fn purge_transfers_until(&mut self, until_timestamp: i64) -> Result<(), Error> {
+        self.state
+            .storage
+            .purge_transfers_until(until_timestamp)
+            .await
+            .map_err(|_| Error::StorageError)
+    }
+
+    pub async fn transfers_since(
+        &mut self,
+        since_timestamp: i64,
+    ) -> Result<Vec<drop_storage::types::Transfer>, Error> {
+        self.state
+            .storage
+            .transfers_since(since_timestamp)
+            .await
+            .map_err(|_| Error::StorageError)
+    }
+
```

Comment:
Minor: to reduce wrapping you can just provide getter `pub fn storage(&self) -> &Storage` and delegate to it down the line

Reply:
Good catch on the `mut`, it remained from when it originally was an `Arc<Mutex<...>>`, As for the layers, i think id like to keep the ffi layer function clean, and leave the processing in `Service`